### PR TITLE
Adding Card variation - Citation

### PR DESCRIPTION
**Pull request by iondrimba**
**CHANGELOG** :memo:

* Added new card variation `card-citation`
* Removed unused card demos
* Removed unused card styles
* Updated card documentation

* This new variation will have limited text like the image below

##### Content style
- I choose not to make an isolated style for the text decoration:
captura de tela 2018-02-14 as 3 57 51 pm

I think we can isolate this style when we start using in more than one place

captura de tela 2018-02-14 as 6 16 26 pm

captura de tela 2018-02-14 as 6 16 32 pm

### CODE REVIEW

**Comment by felipeorlando** [comment]
@iondrimba just a question...

Why to not use a modifier for citation card, like `.card--citation`?

File: src/scss/components/_card-citation.scss [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  &__title {
+    color: get-color(blackNeutral);
+    display: flex;
+    flex-wrap: wrap;
+    font-size: inherit;
+    font-weight: get-weight(normal);
+    margin-bottom: 0;
+
+    strong {
+      font-weight: get-weight(bold);
```
Comment (eduardojmatos) [inline src/scss/components/_card-citation.scss]: `strong` is already `bold`: https://github.com/getninjas/gaiden/blob/master/src/scss/generic/_reset.scss#L11-L13

File: src/scss/components/_card-citation.scss [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }
+  }
+
+  &__text {
+    border-left: 2px solid get-color(offWhite);
+    display: block;
+    padding-left: 12px;
+  }
+
+  .link {
```
Comment (eduardojmatos) [inline src/scss/components/_card-citation.scss]: This `.link` could be a `.card-citation__link`, obeying the BEM methodology.

File: src/scss/generic/_links.scss [hunk 3]
```diff
@@ -1,8 +1,10 @@
-a {
+a,
+.link {
```
Comment (eduardojmatos) [inline src/scss/generic/_links.scss]: Nice. We hadn't a class to do this :)

File: src/scss/tools/mixins/_flex.scss [hunk 4]
```diff
@@ -22,11 +22,6 @@
     $fgBoxflex: nth($fg, 1);
   }
 
-  -moz-box-flex: $fgBoxflex;
```
Comment (eduardojmatos) [inline src/scss/tools/mixins/_flex.scss]: 👏

**Comment by iondrimba** [comment]
@felipeorlando I chose not to go with just a card modifier because we have three variations of cards.
The _card.scss file is getting big and hard to maintain.

**Comment by iondrimba** [comment]
@eduardojmatos Change requests made! I also remover the flex mixin. Looked into the codebase and we are not using it anymore.

File: docs/demo/gaiden-css/components/card/citation.html [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+    body {
+      background-color: #f9f9f9;
+    }
+
+  </style>
+</head>
+
+<body>
+  <div class="container">
+    <a class="col-small-6 space-box-small" href="/reformas-e-reparos/pintor/sp/sao-paulo/orcamentos/5386413-preciso-de-profissional-qualificado">
```
Comment (iondrimba) [inline docs/demo/gaiden-css/components/card/citation.html]: IMPORTANT CHANGE: I move the `<a>` from inside de article to outside.

Comment (iondrimba) [reply docs/demo/gaiden-css/components/card/citation.html]: @felipeorlando @eduardojmatos

File: src/scss/components/_card.scss [hunk 6]
```diff
@@ -82,29 +67,18 @@
...
-
   &__content {
     color: get-color((color: blackNeutral, opacity: .6));
     font-size: 14px;
     font-style: italic;
-
-    .card__header + & {
-      margin-top: 16px;
-    }
+    width: 100%;
```
Comment (iondrimba) [inline src/scss/components/_card.scss]: @felipeorlando @eduardojmatos Testing with diferent text sizes, required to explicit set the width

File: src/scss/components/_card-citation.scss (hunk 1 above)
Comment (iondrimba) [reply src/scss/components/_card-citation.scss]: @eduardojmatos  Nice!

File: docs/demo/gaiden-css/components/card/citation.html (hunk 5 above)
Comment (felipeorlando) [reply docs/demo/gaiden-css/components/card/citation.html]: 🥇

File: src/scss/components/_card-citation.scss [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    display: flex;
+    flex-wrap: wrap;
+    font-size: inherit;
+    font-weight: get-weight(normal);
+    margin-bottom: 0;
+
+    strong,
+    small {
+      font-size: inherit;
+      margin-right: 4px;
```
Comment (williamcosta) [inline src/scss/components/_card-citation.scss]: @iondrimba só um detalhe, esse `margin-right` do strong, está gerando um espaço entre a palavra e os dois pontos (:), ficando meio estranho.

Comment (iondrimba) [reply src/scss/components/_card-citation.scss]: @williamcosta Feito 👍

**Pull request merged**